app/models/stocks/price.rb

```diff
@@ -0,0 +1,35 @@
+require "json"
+require "redis"
+
+class Stocks::Price
+  REDIS_HOST = "redis://redis:6379/"
+
+  attr_accessor :open, :high, :low, :close, :volume,
+    :adj_high, :adj_low, :adj_close, :adj_open, :adj_volume,
+    :split_factor, :dividend, :symbol, :exchange, :date
+
+  def assign_attrs(attrs = {})
+    attrs.each_pair do |k, v|
+      instance_variable_set("@#{k}", v) if v.present? && respond_to?(k)
+    end
+  end
+
+  def initialize(attrs = {})
+    assign_attrs(attrs) unless attrs.blank?
+  end
```

Comment:
These can be replaced with `ActiveModel::Model` , which should give us another handful of benefits as well:

https://guides.rubyonrails.org/active_model_basics.html#model